target/linux/ath79/dts/ar9331_arduino_yun.dts

```diff
@@ -0,0 +1,223 @@
+				pinctrl-single,function-mask = <0x1>;
+				#pinctrl-cells = <2>;
+
+				enable_gpio26_gpio27: pinmux_enable_gpio26_gpio27 {
+					pinctrl-single,bits = <0x0 0x40000 0x40000>;
+				};
+			};
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan {
+			label = "yun:blue:wlan";
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		usb {
+			label = "yun:white:usb";
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+			trigger-sources = <&hub_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		config {
+			label = "config";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_spi_enable {
+			gpio-export,name = "yun:oe:spi";
+			gpio-export,output = <0>;
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_handshake_enable {
+			gpio-export,name = "yun:oe:hs";
+			gpio-export,output = <0>;
+			gpios = <&gpio 22 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_uart_enable {
+			gpio-export,name = "yun:oe:uart";
+			gpio-export,output = <0>;
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	reg_usb_vbus: reg_usb_vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+	};
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&jtag_disable_pins>;
+};
+
+&pinmux_extended {
+	pinctrl-names = "default";
+	pinctrl-0 = <&enable_gpio11>;
+};
+
+&pinmux_bootstrap {
+	pinctrl-names = "default";
+	pinctrl-0 = <&enable_gpio26_gpio27>;
+};
+
+&usb {
+	status = "okay";
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+	dr_mode = "host";
+	vbus-supply = <&reg_usb_vbus>;
+
+	port@1 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+		#trigger-source-cells = <0>;
+
+		hub_port1: port@1 {
+			reg = <1>;
+			#trigger-source-cells = <0>;
+		};
+	};
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x40000 0x10000>;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xf90000>;
+			};
+
+			partition@fe0000 {
+				label = "nvram";
+				reg = <0xfe0000 0x10000>;
+			};
+
+			art: partition@ff0000 {
+				label = "art";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	mtd-mac-address = <&art 0x1002>;
+	mtd-mac-address-increment-byte = <3>;
+	mtd-mac-address-increment = <0x8>;
```

Comment:
I assume you looked for direct sources of the address already?

Reply:
It's complicated. Yun uses bit mask to generate ethernet MAC address.
https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/arch/mips/ath79/mach-arduino-yun.c#L89
In my case `mtd-mac-address-increment[-byte]` worked well because wifi mac was xx:xx:xx:f4:xx:xx. The eth mac will be xx:xx:xx:fc:xx:xx in this case.
But it seems not all devices have mac address with 29th bit off.
https://github.com/arduino/openwrt-arduino-packages/commit/a2daabefe61aa22c0d882b2fded1f5dcff64f803
I assume the author made this commit becuase their device had wifi mac with 29th bit on.
So adding some offset won't have the same effect of masking certain bit.
Currently I'm thinking of making a general version of macaddr_setbit_la() and use it in 02_network.
https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/functions/system.sh#L154
But I'm not sure yet if I should follow ar71xx (wlan0 might have same mac address with eth0) or lede-yun (two address must be different)